Modules/Sources/WordPressUI/Views/Users/Components/UserProfileImage.swift

```diff
@@ -2,36 +2,42 @@ import SwiftUI
 
 struct UserProfileImage: View {
 
-    private let size: CGSize
+    private let size: CGFloat
 
-    private let url: URL
+    private let url: URL?
 
-    init(size: CGSize, email: String) {
-        self.size = size
-        self.url = URL(string: "https://gravatar.com/avatar/58fc51586c9a1f9895ac70e3ca60886e?size=256")!
-    }
-
-    init(size: CGSize, url: URL) {
+    init(size: CGFloat, url: URL?) {
         self.size = size
         self.url = url
     }
 
     var body: some View {
-        AsyncImage(
-            url: self.url,
-            content: { image in
-                image.resizable()
-                    .frame(width: size.height, height: size.height)
-                    .aspectRatio(contentMode: .fit)
-                    .clipShape(.rect(cornerRadius: 4.0))
-            },
-            placeholder: {
-                ProgressView().frame(width: size.height, height: size.height)
-            }
-        )
+        if let url {
+            AsyncImage(
```

Comment:
Please replace this with `AvatarView`. `AsyncImage` should basically [never be used](https://medium.com/@alex_persian/the-case-against-asyncimage-22a073044746).

`AvatarView` or `AsyncImageView` are currently unavailable in `WordPressUI` as they are part of the app target. But `UserProfile*` don't belong to `WordPressUI` as it's designed for reusable components. It should be small and should compile fast as, in the long term, more modules will depend on it.

I suggest the following changes:
- Create a `WordPressMedia` framework (or similar name) with `ImageDownloader` – other media-related stuff could go there later too.
- Move `AvatarView` and `AsyncImageView` to either `WordPressUI` or `WordPressMedia`
- (If you want to keep the previews) Create a separate module for user management.

If we keep adding everything to `WordPressUI`, it will eventually stop working well for Xcode Previews as these tend to become unusable with 10K+ lines in a module. The only long-term option if we want to facilitate Xcode Previews is to create feature-based modules (aka app micro modules). Since we are now fully in the SPM world, it should be feasible, so it might be a good time to start.

Reply:
Sorry, I forgot to reply to your original comment about `AsyncImage`.

Shall I just move the new user management code to the app target for now? So that the `AvatarView` and `ImageDownloader` stuff (I don't think we need it for this particular feature) will be available. /cc @jkmassel

Reply:
Let's address it separately. I'm going to add the approval as is.